ui/e2e/explorer/workflow/variables.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { createNamespace, deleteNamespace } from "../../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { createFile } from "e2e/utils/files";
+import { createWorkflowVariables } from "../../utils/variables";
+import { faker } from "@faker-js/faker";
+import { waitForSuccessToast } from "./utils";
+
+let namespace = "";
+let workflow = "";
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+  workflow = faker.system.commonFileName("yaml");
+  await createFile({
+    name: workflow,
+    namespace,
+    type: "workflow",
+    yaml: "direktiv_api: workflow/v1\nstates:\n- id: noop\n  type: noop",
+  });
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+});
+
+test("bulk delete workflow variables", async ({ page }) => {
+  // Create 3 test variables
+  await createWorkflowVariables(namespace, workflow, 3);
+
+  await page.goto(`/n/${namespace}/explorer/workflow/settings/${workflow}`);
+
+  // Check first checkbox and click delete
+  await page.getByTestId("item-name").getByRole("checkbox").first().check();
+  await page.getByRole("button", { name: "Delete" }).click();
+  await expect(
+    page.getByText(`Are you sure you want to delete variable`, { exact: false })
+  ).toBeVisible();
+  await page.getByRole("button", { name: "Cancel" }).click();
+
+  // Check second checkbox and click delete
+  await page.getByTestId("item-name").getByRole("checkbox").nth(0).check();
+  await page.getByTestId("item-name").getByRole("checkbox").nth(1).check();
+  await page.getByRole("button", { name: "Delete" }).click();
+  await expect(
+    page.getByText("Are you sure you want to delete 2 variables?", {
+      exact: true,
+    })
+  ).toBeVisible();
+  await page.getByRole("button", { name: "Cancel" }).click();
+
+  // Check third checkbox and click delete
+  await page.getByTestId("item-name").getByRole("checkbox").nth(0).check();
+  await page.getByTestId("item-name").getByRole("checkbox").nth(1).check();
+  await page.getByTestId("item-name").getByRole("checkbox").nth(2).check();
+  await page.getByRole("button", { name: "Delete" }).click();
+  await expect(
+    page.getByText("Are you sure you want to delete all variables?", {
+      exact: true,
+    })
+  ).toBeVisible();
+
+  // Confirm deletion
+  await page.getByRole("button", { name: "Delete" }).click();
+  await waitForSuccessToast(page);
+
+  await expect(page.getByTestId("item-name")).toHaveCount(0);
+});
```

Comment:
Very good testing here. Especially with checking the different modal messages. Could we add one small check to make sure a single delete works as well? Maybe just add 4 variable instead of 3 and in the beginning of this test you delete the first and check if the list updated?